prymer/primer3/primer3_parameters.py

```diff
@@ -204,6 +220,15 @@ def __post_init__(self) -> None:
             raise TypeError("Probe melting temperatures and GC content must be floats")
         if self.probe_max_dinuc_bases % 2 == 1:
             raise ValueError("Max threshold for dinucleotide bases must be an even number of bases")
+        default_thermo_max: float = self.probe_tms.min - 10.0
+        thermo_max_fields = [
+            "probe_max_self_any_thermo",
+            "probe_max_self_end_thermo",
+            "probe_max_hairpin_thermo",
+        ]
+        for field in fields(self):
+            if field.name in thermo_max_fields and getattr(self, field.name) is None:
+                object.__setattr__(self, field.name, default_thermo_max)
```

Comment:
Is there a reason to use getattr/setattr here instead of just having three lines that look like:

```python
self.probe_max_self_any_thermo = self.probe_max_self_any_thermo if self.probe_max_self_any_thermo is not None else default_thermo_max
```

Reply:
Rule of three?

Subjective preference? (The ternary is 133 characters without indentation, so it's going to get a gnarly linewrap or need a `#noqa` pragma)

A slightly simpler implementation might look like:
```py
for fieldname in thermo_max_fields:
    if getattr(self, fieldname) is None:
        object.__setattr__(self, fieldname, default_thermo_max)
```

Reply:
Sure ... but if we're counting lines then the 5 lines do declare the list of three attributes counts too.  But more importantly, even if it's slightly more code, writing it out using symbols vs. strings is more robust to refactoring, find-usages, etc.

Reply:
I'd like to mark this as resolved since we covered it in an internal meeting the other day: this is a frozen dataclass so a little extra footwork is required to set some attributes based on other attributes in a `__post_init()__`